### ISSUE TITLE
이강현_중간고사

### ISSUE THREAD

**Issue by leeganghyun97** [KimSeungJun21/Team_project#1]
[2024196227_이강현_중간과제.zip](https://github.com/user-attachments/files/17742975/2024196227_._.zip)

ㅇ정성평가(15점) 1. 데이터 수집: 설명변수를 다양하게 수집하였네요. 다만, 왜 해당 변수를 수집하였는지, 어떤 문헌을 통해 이들 변수를 수집하였다를 언급하거나 아니면 해당 레퍼런스가 없네요. 물론 금융도메인이 익숙치 않고, 시간이 부족해서 문헌조사가 어려웠던 점은 이해합니다. 그러나, 최소 몇 개의 변수라도 조사를 통해 수집되었다는 근거를 남겼다면 좋았을 것 같습니다. 나아가 실제 논문들 보면 관련 인용과 함께 Relation direction을 +, - 인지까지 상세하게 기재하는 경우도 많습니다. 그만큼 신뢰감 있게 변수들을 활용하였다를 보여주는게 중요합니다. 미국 금리라고 하면, 미국 정책 금리인가요? 아니면 시장금리일까요? 변수에 대한 내용 설명도 필요해요. 만약, 기준금리라면 거의 몇 개월 연속으로 고정되어 가는 경우가 많습니다. 최근 코로나 기간때만 해도 빠르게 인상되다가 최근 몇 년간 고정되어 있기 때문에 모델이 예측하고자 하는 타겟 변수에 미치는 영향이 거의 없기 때문입니다.그래서 변화가 미미하면 모델은 학습 중 해당 변수를 무시하는 경향이 생길 수 있습니다. 이에 반해 시장금리를 여러 개를 활용하는 건 매우 좋네요. 시장금리는 시장에서 자율적으로 매일 바뀌는 금리니깐요. 그런데 보통 경제변수를 예측할 때, 시장금리는 국고채의 단기물, 장기물 이렇게 만기에 따라서 몇가지만 활용합니다. 서로 상관관계가 높은 시장금리 변수들을 여러 개 투입하면 다중공선성만 투입돼서 모델의 복잡성은 더 올라가니깐요. 그러나, 각각 만기가 장기, 단기면 서로 상관성이 덜합니다. 그래서 시장금리 데이터를 모집할때는 항상 만기 몇 년짜리인지를 써주는게 일반적입니다. 그외, 변수별로 가중치를 달리 적용해서 변수를 활용하는 법은 참신하네요. 아마 Feature importance를 산출해서 가중치를 적용한건가요? 가중치 활용 자체는 Novelty가 좋아보여요. 그런데, 왜 어떤 근거로 가중치를 적용한 건지 설명이 안된다면, 누가봐도 동 방법론에 대해 의문점이 생깁니다. 코드상에 제가 위에 언급한 내용까지 일일이 서술해야 되나 생각이 들 수 있겠죠. 금번 중간과제는 단순, Kaggle Competition이 아닙니다. 하나의 분석 보고서를 만드는 거란 걸 제가 강조한바 있습니다. Home work이 아닌 중간고사를 대체하는 비중입니다. 2. 데이터 전처리: 결측치를 확인하는 것은 좋습니다. 현재 CPI나 여타 데이터의 경우 결측치는 없을겁니다. 관리가 잘 되는 데이터이니깐요. 다만 시계열의 경우 결측치를 날려버리면 시계열 의존성이 끊겨버리는 문제가 발생해요. 반드시 보간해야 해요. 또한, 이상치에 대해서 IQR이나 Z-score 등으로 한번 확인해보는 것도 좋겠죠? 또한, 상관관계 분석, 통계 요약이나 산점도 차트 등 여러가지 시각화를 해보는 것도 좋습니다. 데이터 분포가 대략 어떻게 되는지를 잘 살펴봐야 해당 데이터를 기반으로 구축한 인공지능 모델을 신뢰 할 수 있습니다. Scale 조정은 훌륭합니다. Min-Max를 썼군요. 시차(Lagged)변수도 좋습니다. Lagged 변수에 대해서도 추가해본 것은 좋은 시도입니다. 실제로 이런 방법으로 모델링하는 VAR 모델이 있습니다. 다만, 아래와 같이 실제 문헌들을 인용하여 효과가 있을 것이다라고 전제하여 시도해보면 더 좋겠죠? Stock, J.H., & Watson, M.W. (2001). "Vector Autoregressions," Journal of Economic Perspectives, 15(4): 101–115. 만약 학부생 과제였다면 이런 근거가 없어도 이해할겁니다. 다만, 대학원 과제이니 좀더 신뢰할 수 있는 접근임을 강조하면 좋겠죠. 훌륭한 시도이긴 합니다. 3. 모델 구현: Batch normalization 과 Dropout rate을 동시에 써서 정규화를 수행했군요. 그리고, 2fold CV 방식으로 LSTM 모델을 튜닝했군요. 훌륭해요. 다만, 튜닝한 하이퍼파라미터 종류가 너무 적습니다. 레이어, 노드 외에도 dropout rate, learning rate, batch size, epoch 등 많습니다. 튜닝 과정이 번거롭고 시간도 많이 걸리는 걸 압니다. 그래도 튜닝 과정은 정말 중요해요. 연구의 아이디어에 있어서는 중요한 포인트는 아니지만, 나중에 독자가 우리의 모델을 그대로 재현해서 쓴다는 것을 알아야 합니다. 그런데, 하이퍼파라미터를 다르게 하면 더 성능이 좋아지더라? 그러면 신뢰감이 완전 떨어지겠죠. 그래서 보통 튜닝을 하고 그 과정을 APPENDIX에 붙입니다. 또한, Dense layer도 보통 2-3개 정도 포함합니다. 앞에서 LSTM 레이어로 순차적 의존성을 제대로 학습시켜 줄수 있지만, 결국 LSTM 이 추출한 순차적 특징들을 비선형적으로 변환하여 더 추상적인 수준으로 만들어 내는 것은 Dense layer의 역할입니다. 그외 벤치마크 모델도 활용해서, 내가 구현한 RNN 계열 모델이 과연 타 모델 대비 얼마나 예측 성능이 좋은가 떨어지는가도 비교해보면 좋겠죠. MSE, RMSE 지표는 결국 상대적인 지표입니다. 실제 값이 scale이 크다면 MSE도 크겠죠. 그래서 에러 수준이 어떻다라는 것은 결국 타 모델과의 상대적 비교하는 방법이 가장 효율적이죠 제가 유의사항으로 Competition 결과와 코드를 compile 했을 때, MSE가 비교가 가능하게 거듭 강조드린 바 있습니다. 이런 유의사항 부분대로 구현이 잘 안되어 있습니다. 제가 이강인 학생 코드에 MSE를 찍어내게 코드를 직접 구현할 수도 있지만, 그렇게는 안하겠습니다. 그 외 각 과정에 대한 본인의 해석들이 전혀 없어요. 다시한번 강조하지만, 금번 중간 과제는 단순히 예측 성능 높일수록 고득점 받는 Competition이 아닙니다. 분석 보고서와 같아요. 아마 시간이 부족해서 이런 부분들이 미흡했을 거라 생각합니다. 그럼에도, 중간 과제 수행하느라 다시한번 고생 많았어요^^
11월 3일 오후 3:43에 박상진 / Park, Sangjin이(가) 작성
ㅇ정량평가 16점 kaggle 1.8 실제 2.3 재현성이 0.5 이내면 ok
[Uploading 2024196227_이강현_중간과제.zip…]()
